src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs

```diff
@@ -66,18 +69,29 @@ public class ForkchoiceUpdatedHandler : IForkchoiceUpdatedHandler
         _mergeSyncController = mergeSyncController;
         _beaconPivot = beaconPivot;
         _peerRefresher = peerRefresher;
+        _specProvider = specProvider;
         _logger = logManager.GetClassLogger();
     }
 
-    public Task<ResultWrapper<ForkchoiceUpdatedV1Result>> Handle(ForkchoiceStateV1 forkchoiceState, PayloadAttributes? payloadAttributes)
+    public Task<ResultWrapper<ForkchoiceUpdatedV1Result>> Handle(ForkchoiceStateV1 forkchoiceState, PayloadAttributes? payloadAttributes, int version)
+    {
+        Block? newHeadBlock = GetBlock(forkchoiceState.HeadBlockHash);
+        ResultWrapper<ForkchoiceUpdatedV1Result>? payloadUpdateResult = ApplyForkchoiceUpdate(newHeadBlock, forkchoiceState, payloadAttributes);
+        return Task.FromResult(
+            ValidateAttributes(payloadAttributes, version) ??
+            payloadUpdateResult ??
+            StartBuildingPayload(newHeadBlock!, forkchoiceState, payloadAttributes)
+        );
```

Comment:
This is the main change:
- run ApplyForkchoiceUpdate, 
- only then validate params and fork, return error if found, 
- or error from ApplyForkchoiceUpdate 
- or anything else returned by StartBuildingPayload)

@jflo @MarekM25